boards/risc-v/esp32c3/esp32c3-devkit/src/esp32c3_reset.c

```diff
@@ -55,6 +58,23 @@
 
 int board_reset(int status)
 {
+#ifdef CONFIG_BOARD_ASSERT_RESET_VALUE
+  syslog(LOG_INFO, "reboot status=%d\n", status);
+
+  switch (status)
+    {
+      case EXIT_SUCCESS:
+        up_shutdown_handler();
+        break;
+      case CONFIG_BOARD_ASSERT_RESET_VALUE:
+        break;
+      default:
+        break;
+    }
+#else
+  up_shutdown_handler();
+#endif
```

Comment:
```suggestion
  syslog(LOG_INFO, "reboot status=%d\n", status);

  switch (status)
    {
      case EXIT_SUCCESS:
        up_shutdown_handler();
        break;
      case CONFIG_BOARD_ASSERT_RESET_VALUE:
        break;
      default:
        break;
    }
#endif
```
`BOARD_ASSERT_RESET_VALUE` depends on `BOARDCTL_RESET`, so the `#else` clause will never really happen and can be removed.